### PR TITLE
Allow blackhole to delay response

### PR DESCRIPTION
**Pull request by blt**
### What does this PR do?

This commit allows lading HTTP blackhole to delay a response to
the sender, intention being to allow for the simulation of latent
network connections to a sink location.

### CODE REVIEW

**Comment by blt** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/DataDog/lading/1012?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1012** <a href="https://app.graphite.dev/github/pr/DataDog/lading/1012?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1011** <a href="https://app.graphite.dev/github/pr/DataDog/lading/1011?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @blt and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: lading/src/blackhole/http.rs [hunk 1]
```diff
@@ -98,6 +102,9 @@
     /// raw array of bytes if the `raw_bytes` body variant is selected
     #[serde(default)]
     pub raw_bytes: Vec<u8>,
+    /// delay to add before making a response
+    #[serde(default = "default_response_delay")]
+    pub response_delay: Duration,
```
Comment (GeorgeHahn) [inline lading/src/blackhole/http.rs:107]: IIRC the default deserialization for `Durations` is pretty bad (map with `sec`, `nanos`?) Could you verify how this shows up in config? I'm not sure how much attention this is worth, but accepting human time units or fractional seconds seems preferable.

Comment (blt) [reply lading/src/blackhole/http.rs:107]: Ah that's a good point. I always forget how wonky this is. I'll make it an explicit millisecond setting.

File: lading/src/blackhole/http.rs [hunk 2]
```diff
@@ -98,6 +102,9 @@ pub struct Config {
     /// raw array of bytes if the `raw_bytes` body variant is selected
     #[serde(default)]
     pub raw_bytes: Vec<u8>,
+    /// delay to add before making a response
+    #[serde(default = "default_response_delay_millis")]
+    pub response_delay_millis: u64,
```
Comment (goxberry) [inline lading/src/blackhole/http.rs:107]: Do you foresee possibly changing the default response delay?

**Review by goxberry — APPROVED** [review]

**Comment by blt** [comment]
### Merge activity

* **Oct 2, 8:04 PM EDT**: @blt merged this pull request with [Graphite](https://app.graphite.dev/github/pr/DataDog/lading/1012).

File: lading/src/blackhole/http.rs (hunk 2 above)
Comment (scottopell) [reply lading/src/blackhole/http.rs:107]: Interesting question, it seems useful to make this a little closer to real-world conditions where you'll never have a 0ms rtt. 
Asked some internal teams to see if we can get a better default

Comment (goxberry) [reply lading/src/blackhole/http.rs:107]: The original motivation for the question was a bit more mundane: if the default will always be zero, `#[serde(default = "default_response_delay_millis"]` could be replaced with `#[serde(default)]`. However, delegating the default value to a function makes changing that default value much easier.

**Pull request merged**